### DISCUSSION TITLE
Add additional information to the output

### DISCUSSION THREAD

**Discussion by mcg1103** [eulerto/wal2json#269] [Ideas]
I am currently using the output of this plugin to update a search system.  So any changes to a record then I update a solr document.  This is working find.
I was considering using the output to log changes to the database.  Track changes to the database similar to the trigger based audit log or just the statement logging built into postgres.

We use a connection pool so all database users are the same user.  We would have to set a "current_setting" 
`set local "app.userid" = 2;
select current_setting('app.userid', true); `
Could also do a "set role bob" on the connection prior to running the query an set the connection user on the response.

I looked into setting the user in the json via this system and I could not find where the connections role was getting passed in on the transaction.... it may not be so this is moot, but it would be amazing if this was possible.

**Comment by eulerto** [comment]
Logical replication only streams data. Environment properties / settings are not replicated. One solution is to use a logical decoding message. Write a transactional message using the `pg_logical_emit_message` function. See [Examples](https://github.com/eulerto/wal2json#examples).
